### ISSUE TITLE
bug: PDF with multiple columns not partitioned correctly

### ISSUE THREAD

**Issue by ys-1234** [Unstructured-IO/unstructured#1336]
**Describe the bug**
When partition PDF file containing multiple columns, the partitioned contents are shuffled. For example, the left-most column appears after the right-most column.

**To Reproduce**
```
elements = partition(pdf_file_path)
new_file_path = dir / pathlib.Path(file_path).with_suffix(".txt").name
    with open(new_file_path, "w") as text_file:
        text_file.write("\n".join([str(el) for el in elements]))
```

**Expected behavior**
The elements should be returned in a readable sequence.

**Screenshots**
Here is how the pdf input looks like. I also attached the actual file and the output log.
<img width="773" alt="Screenshot 2023-09-07 at 3 05 49 PM" src="https://github.com/Unstructured-IO/unstructured/assets/106637777/387645ca-c8dd-492f-badd-f3bfb00ae123">

Input file: [dc23sint.pdf](https://github.com/Unstructured-IO/unstructured/files/12553720/dc23sint.pdf)
Partitioned element joined with '\n':[dc23sint.txt](https://github.com/Unstructured-IO/unstructured/files/12553721/dc23sint.txt)

**Environment Info**
Output of `python scripts/collect_env.py`:
```
OS version:  macOS-10.16-x86_64-i386-64bit
Python version:  3.8.6
unstructured version:  0.10.5
unstructured-inference is not installed
pytesseract version:  0.3.10
Torch version:  1.13.1
Detectron2 is not installed
PaddleOCR is not installed
Libmagic version: file-5.41
magic file from /usr/share/file/magic
Traceback (most recent call last):
  File "/Users/ys/Downloads/c2k/chunk_revamp_results/collect_env.py", line 251, in <module>
    main()
  File "/Users/ys/Downloads/c2k/chunk_revamp_results/collect_env.py", line 243, in main
    libreoffice_version = get_libreoffice_version()
  File "/Users/ys/Downloads/c2k/chunk_revamp_results/collect_env.py", line 171, in get_libreoffice_version
    result = subprocess.run(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'libreoffice'
```
**Additional context**
N/A

**Comment by walsha2** [comment]
@ys-1234 You appear to be using `unstructured version:  0.10.5` 

Can you try one of the more recent versions, like `0.10.12`.

I tested your PDF using `0.10.12` and got much better results, with the columns in the correct order:  [out.txt](https://github.com/Unstructured-IO/unstructured/files/12553955/out.txt)

**Comment by cragwolfe** [comment]
Yes, per @walsha2 's comment, there was a big improvement on ordering released in `0.10.6` that does especially well with clean divisions between columns as the example in this issue has.

For further improvements, keep an eye on https://github.com/Unstructured-IO/unstructured/issues/1233
and
https://github.com/Unstructured-IO/unstructured/issues/1209 .

Closing since it looks like the issue is resolved for this particular document. Thanks for testing it with the recent version, @walsha2 !
